projects/robots/abb/irb/controllers/inverse_kinematics/inverse_kinematics.py

```diff
@@ -30,58 +30,24 @@
     sys.exit('The "ikpy" Python module version is too old. '
              'Please upgrade "ikpy" Python module to version "3.0" or newer with this command: "pip install --upgrade ikpy"')
 
-# Create the arm chain.
-# The constants below have been manually extracted from the Irb4600-40.proto file, looking at the HingeJoint node fields.
-# The chain should contain the "E motor" bone, because this bone defines the hand position.
-armChain = Chain(name='arm', links=[
-    OriginLink(),
-    URDFLink(
-        name="A motor",
-        bounds=[-3.1415, 3.1415],
-        translation_vector=[0, 0, 0.159498],
-        orientation=[0, 0, 0],
-        rotation=[0, 0, 1]
-    ),
-    URDFLink(
-        name="B motor",
-        bounds=[-1.5708, 2.61799],
-        translation_vector=[0.178445, -0.122498, 0.334888],
-        orientation=[0, 0, 0],
-        rotation=[0, 1, 0]
-    ),
-    URDFLink(
-        name="C motor",
-        bounds=[-3.1415, 1.309],
-        translation_vector=[-0.003447, -0.0267, 1.095594],
-        orientation=[0, 0, 0],
-        rotation=[0, 1, 0]
-    ),
-    URDFLink(
-        name="D motor",
-        bounds=[-6.98132, 6.98132],
-        translation_vector=[0.340095, 0.149198, 0.174998],
-        orientation=[0, 0, 0],
-        rotation=[1, 0, 0]
-    ),
-    URDFLink(
-        name="E motor",
-        bounds=[-2.18166, 2.0944],
-        translation_vector=[0.929888, 0, 0],
-        orientation=[0, 0, 0],
-        rotation=[0, 1, 0]
-    )
-])
-
 # Initialize the Webots Supervisor.
 supervisor = Supervisor()
 timeStep = int(4 * supervisor.getBasicTimeStep())
 
+# Create the arm chain from the URDF
+filename = None
+with tempfile.NamedTemporaryFile(suffix='.urdf', delete=False) as file:
+    filename = file.name
+    file.write(supervisor.getUrdf())
```

Comment:
I am getting an error here with Python 3.8:
```
Traceback (most recent call last):
  File "inverse_kinematics.py", line 41, in <module>
    file.write(supervisor.getUrdf())
  File "C:\Program Files\Python38\lib\tempfile.py", line 474, in func_wrapper
    return func(*args, **kwargs)
TypeError: a bytes-like object is required, not 'str'
WARNING: 'inverse_kinematics' controller exited with status: 1.
```
I could fix it with:
```suggestion
    file.write(supervisor.getUrdf().encode('utf-8'))
```